File: app/src/main/java/br/com/santiago/teste/soccernews/data/local/AppDatabase.java
package br.com.santiago.teste.soccernews.data.local;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import br.com.santiago.teste.soccernews.domain.News;

@Database(entities = {News.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract NewsDao newsDao();
}

File: app/src/main/java/br/com/santiago/teste/soccernews/data/local/AppDatabase.kt
package br.com.santiago.teste.soccernews.data.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import br.com.santiago.teste.soccernews.domain.News

@Database(entities = [News::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun newsDao(): NewsDao

    companion object {
        private lateinit var INSTANCE: AppDatabase


        fun getDatabase(context: Context): AppDatabase {
            if (!Companion::INSTANCE.isInitialized) {
                synchronized(AppDatabase::class) {
                    INSTANCE = Room.databaseBuilder(
                        context,
                        AppDatabase::class.java,
                        "soccer-news"
                    ).allowMainThreadQueries()
                        .build()
                }
            }
            return INSTANCE
        }
    }
}